sdks/wasm/DebuggerTestSuite/Tests.cs

```diff
@@ -1243,6 +1247,141 @@ public async Task InspectLocalsForStructInstanceMethod (bool use_cfo)
 							label: "this#0");
 
 				});
+
+		[Fact]
+		public async Task SetBreakpointBeforeInspectorReady () {
+			var insp = new Inspector ();
+			
+			var scripts = SubscribeToScripts (insp);
+			await Ready (); 
+
+			using (var cts = new CancellationTokenSource()) {
+				cts.CancelAfter (60 * 1000);
+				var uri = new Uri ($"ws://{TestHarnessProxy.Endpoint.Authority}/launch-chrome-and-connect");
+				using var loggerFactory = LoggerFactory.Create(
+					builder => builder.AddConsole().AddFilter(null, LogLevel.Trace));
+				using (var client = new InspectorClient (loggerFactory.CreateLogger<Inspector>())) {
+					await client.Connect (uri, insp.OnMessage, async token => {
+						Task[] init_cmds = {
+							client.SendCommand ("Profiler.enable", null, token),
+							client.SendCommand ("Runtime.enable", null, token),
+							client.SendCommand ("Debugger.enable", null, token),
+							client.SendCommand ("Runtime.runIfWaitingForDebugger", null, token),
+							insp.WaitFor ("ready"),
+						};
+
+						insp.On("Debugger.breakpointResolved", async (args, c) => { 
+							Console.WriteLine ("BREAKPOINT RESOLVED");
+							event_order.Add ("breakpointResolved");
+						});
+
+						Console.WriteLine ("waiting for the debugger to be ready");
+						await init_cmds[2];
+						Assert.False (scripts.Values.Contains("dotnet://debugger-test.dll/debugger-test.cs"));
+						string url = $"file://{Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.Parent.FullName}/debugger-test.cs";
+						Console.WriteLine ("PATH " + url);
+						ctx = new DebugTestContext (client, insp, token, scripts);
+						var bp1_req = JObject.FromObject (new { lineNumber = 5, columnNumber = 2, url = url,});
```

Comment:
Um you can use `SetBreakpoint` here too

Reply:
I can't because the script hasn't been loaded yet. And `SetBreakpoint` pulls from `dicFileToUrl` which hasn't been loaded.